### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

File: src/widgets/buttons.ts [hunk 1]
```diff
@@ -184,19 +189,37 @@ function setName (element, x) {
...
+ *          * ns.foaf('depiction')
+
+ */
+function imagesOf (x: NamedNode, kb: IndexedFormula): any[] {
   var ns = UI.ns
   return kb
     .each(x, ns.sioc('avatar'))
     .concat(kb.each(x, ns.foaf('img')))
     .concat(kb.each(x, ns.vcard('logo')))
+    .concat(kb.each(x, ns.vcard('hasPhoto')))
```
Comment (michielbdejong) [inline src/widgets/buttons.ts]: To align with `findImage`

**Comment by michielbdejong** [comment]
Screenshot 2020-02-19 21 20 44

**Comment by michielbdejong** [comment]
@SharonStrats let's solve the webpack build problem in a separate issue:
https://github.com/solid/solid-ui/issues/213
In the meantime you can check the examples page at: http://solid.github.io/solid-ui/examples/buttons/
Please re-review

File: src/widgets/buttons.ts [hunk 2]
```diff
@@ -24,18 +26,25 @@ const dragAndDrop = require('./dragAndDrop')
...
+  let dom = context && context.dom
   if (!dom && typeof document !== 'undefined') {
     dom = document
   }
   if (dom) {
     var body = dom.getElementsByTagName('body')[0]
-    box = dom.createEvent('div')
-    body.insertBefore(box, body.firstElementChild)
-    context.statusArea = box
+    ;(box as unknown as HTMLElement) = dom.createElement('div')
```
Comment (SharonStrats) [inline src/widgets/buttons.ts]: Having to use 'as' will be something we look to change when we refactor to Typescript completely?

File: src/widgets/buttons.ts [hunk 3]
```diff
@@ -332,10 +373,10 @@ function trySetImage (element, thing, iconForClassMap) {
   return false // we can do better
 }
 
-// ToDo: Also add icons for *properties* like  home, work, email, range, domain, comment,
-//
-
-function setImage (element, thing) { // 20191230a
+/**
```
Comment (SharonStrats) [inline src/widgets/buttons.ts]: We would like to change // to /*  */?

File: src/widgets/buttons.ts (hunk 2 above)
Comment (michielbdejong) [reply src/widgets/buttons.ts]: Ah actually there the cast was unnecessary.

In other places I cast from `HTMLElement | null` to `HTMLElement` to indicate that the code ignores the possibility of `document.createElement('div')` returning null.

**Pull request merged**